src/core/task/Task.ts

```diff
@@ -1117,6 +1118,104 @@ export class Task extends EventEmitter<ClineEvents> {
 		})
 	}
 
+	// Helper methods for synthetic message generation
+
```

Comment:
Task is getting too big, can you keep the new functional implementation in a new file module?